File: Combination Sum IV.py
class Solution(object):
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        nums.sort()
        dyna = [1] + [0]*target
        for i in range(1, target+1):
            for j in nums:
                if j <= i: dyna[i] += dyna[i-j]
                else: break
        return dyna[target]

File: Combination Sum IV.java
public class Solution {
    public int combinationSum4(int[] nums, int target) {
        Arrays.sort(nums);
        int dyna[] = new int[target+1];
        dyna[0] = 1;
        for (int i = 1; i <= target; i++){
            for (int j : nums){
                if (j <= i) dyna[i] += dyna[i-j];
                else break;
            }
        }
        return dyna[target];
    }
}